### PR TITLE
Preparations for Engineblock 5.x as well as Development Environment improvements

### PR DESCRIPTION
**Pull request by DRvanR**
Replaces #22 

---

- Created new `./provision-develop` script to create a development VM
- Updated scripts to enable easier development when creating a dev VM (e.g. [disabling haproxy health checks][1])
- Added documentation covering the development VM
- Remove Profile configuration from EngineBlock, including Vhost config as profile has been removed

This PR is against the engineblock-5.x-dev branch, _not_ master. That is on purpose, as the replacement for profile is still in development.

This PR is made together with https://github.com/OpenConext/OpenConext-engineblock/pull/193 - the one cannot work with the other, and without the new profile app neither may be deployed.

[1]: https://github.com/OpenConext/OpenConext-deploy/commit/e2b57d5bffb7b1e0d5f51908d18d1e78d361d7c4

### CODE REVIEW

**Comment by DRvanR** [comment]
@rjkip please have a look :dash:

File: provision-vm.yml [hunk 1]
```diff
@@ -54,7 +53,7 @@
     - { role: welcome,           tags: ['welcome'] }
     - { role: openconext-common, tags: ['eb','sr'] }
     - { role: engineblock,       tags: ['eb'     ] }
-    - { role: vm_only_bootstrap_ldap_attributes,       tags: ['ldap'     ] }
+#    - { role: vm_only_bootstrap_ldap_attributes,       tags: ['ldap'     ] }
```
Comment (rjkip) [inline provision-vm.yml]: What's this for?

Comment (DRvanR) [inline provision-vm.yml]: Remnant of testing. It is duplicated two lines below and caused the 2nd to consistently fail (which also was ignored). Tried it without the first invocation (commented it out as it is currently), and it seems to keep on working correctly...

Perhaps @thijskh can shed some light on whether or not this is/could be an issue?

File: Vagrantfile [hunk 2]
```diff
@@ -13,13 +13,22 @@ Vagrant.configure("2") do |config|
...
       vb.customize ["modifyvm", :id, "--cpus", "2"]
     end
+
+    if ENV['ENV'] == 'develop'
+        apps.vm.synced_folder "../OpenConext-engineblock", "/opt/openconext/OpenConext-engineblock"
+# The reason to do this is so that we can create writeable dirs in the project under a different user (cache dir etc).
+# This in turn is wanted since that allows us to use additional tooling for Symfony2 that needs the cache for introspection.
+# However, this user does not exist until *after* the ansible provisioning ran - so it can't run till then :(
+# Should be resolved at a later time.
+#         apps.vm.provision :shell, run: "always", inline: "mount -t vboxsf -o uid=engineblock,gid=engineblock opt_engineblock /opt/openconext/OpenConext-engineblock"
```
Comment (rjkip) [inline Vagrantfile]: We should put this in a ticket somewhere and remove this.

File: roles/engineblock/tasks/develop.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Currently not used, see the Vagrant file for the rationale.
+# When used, move all dev specific actions to this file.
```
Comment (rjkip) [inline roles/engineblock/tasks/develop.yml]: > We should put this in a ticket somewhere and remove this.

File: Vagrantfile (hunk 2 above)
Comment (DRvanR) [inline Vagrantfile]: Fair enough, will do.

File: roles/haproxy/templates/haproxy.cfg.j2 [hunk 4]
```diff
@@ -35,7 +35,11 @@ global
     daemon
 
     # turn on stats unix socket
+{% if develop %}
+    stats socket /var/lib/haproxy/stats mode 600 level admin
+{% else %}
     stats socket /var/lib/haproxy/stats
```
Comment (rjkip) [inline roles/haproxy/templates/haproxy.cfg.j2]: What's this for? I'm curious. Perhaps we should document why this is done.

File: roles/mysql/tasks/main.yml [hunk 5]
```diff
@@ -55,3 +55,8 @@
     - database_clients
   tags: mysql
 
+- name: Create development database user
+  mysql_user: name={{ item.name}} host=% password={{ item.password }} priv=*.*:ALL state=present
+  with_items:
+    - "{{ databases.development_user }}"
+  when: develop
```
Comment (rjkip) [inline roles/mysql/tasks/main.yml]: We should document somewhere that the database username and password, when developing, is `development:development`.

File: roles/haproxy/templates/haproxy.cfg.j2 (hunk 4 above)
Comment (DRvanR) [inline roles/haproxy/templates/haproxy.cfg.j2]: This is just haproxy specific, not sure it should be documented (you should not alter functionality you don't understand in the first place :wink: ). The develop only creates a socket that accepts admin commands, which we need to disable health checks on dev, as is done [here][1].

[1]: https://github.com/OpenConext/OpenConext-deploy/blob/development-environment/roles/haproxy/tasks/main.yml#L42

Comment (rjkip) [inline roles/haproxy/templates/haproxy.cfg.j2]: If it makes sense for people that know HAProxy then forget I ever commented on this.

File: roles/mysql/tasks/main.yml (hunk 5 above)
Comment (DRvanR) [inline roles/mysql/tasks/main.yml]: Done in 3aa7831

File: provision-vm.yml (hunk 1 above)
Comment (thijskh) [inline provision-vm.yml]: I asked the original committer and he doesn't know why he did that, he says that it seems wrong and that it's probably just by mistake.

**Comment by DRvanR** [comment]
@rjkip thoughts?

**Comment by rjkip** [comment]
:rocket:

**Pull request merged**